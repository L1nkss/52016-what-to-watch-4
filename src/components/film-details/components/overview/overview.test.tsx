import * as React from "react";
import * as renderer from "react-test-renderer";
import Overview from "@components/film-details/components/overview/overview";

const film = {
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewVideoLink: `img/bohemian-rhapsody.jpg`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Drama`,
  previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  released: 2015,
  id: 2,
  runTime: 198,
  starring: [`One`, `Two`],
  director: `Me`,
  isFavorite: false,
  description: `Text Text Text`,
  scoresCount: 125,
  rating: 0.5
};


describe(`Testing overview component`, () => {
  it(`Component should successfully rendered`, () => {
    const tree = renderer.create(<Overview data={film}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
