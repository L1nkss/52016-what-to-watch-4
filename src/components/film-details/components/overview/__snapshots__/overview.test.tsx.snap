// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing overview component Component should successfully rendered 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      0.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        125
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    Text Text Text
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Me
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        One Two
      </strong>
    </p>
  </div>,
]
`;
